src/libAtomVM/intn.c

```diff
@@ -923,14 +924,25 @@ int intn_parse(
             // TODO: check overflows
             intn_mulmnu(out, out_len, mult, 2, new_out);
             new_out_len = MAX(2, intn_count_digits(new_out, INTN_MUL_OUT_LEN(out_len, 2)));
+            if (UNLIKELY(out_len > INTN_MAX_IN_LEN)) {
+                // we are above the allowed 256 bits, so it is going to be overflow
+                // if still have some room in our buffer, so we are safe
+                return -1;
+            }
+            assert(out_len <= INTN_MAX_RES_LEN);
```

Comment:
This assert seems pointless since the out_length was already just checked. It seems impossible for the asset to ever fail. Doesn’t the redundant check add runtime overhead?

Reply:
The idea was to add an additional check to make sure we are not running out of buffer space. I move it since it was misplaced.